Expensify/App
# Update splash screen logic

Following https://github.com/Expensify/App/issues/5620

### Details

There are several issue with the current splash integrations:

- Multiple `UIViewController` could be mounted at the same time (splash screen & growl notifications, could cause https://github.com/zoontek/react-native-bootsplash/issues/119)
- `bootsplash.hide()` could reject, which is not handled (see https://github.com/zoontek/react-native-bootsplash/issues/264, which happen on tapping notification on iOS 15+)
- It doesn't wait for `NavigationContainer` `onReady` callback to hide.
- It will not hide if something happen during context providers initialisation.

### Fixed Issues

https://github.com/Expensify/App/issues/5620

### Tests


### QA Steps

- Starts the app on Android >= 5, check if the splash screen is visible
- Starts the app on iOS >= 10, check if the splash screen is visible
- Try popping an alert / a native modal once the screen is loaded
- Open the app using a push notification tap (on both iOS and Android)

### Tested On


### Screenshots

#### Web

<img src="https://user-images.githubusercontent.com/1902323/144210429-6d1e1a3d-1b7c-4e5f-8948-8113862ee024.png" width="640" />

#### Mobile Web

<img alt="Screenshot 2021-12-01 at 10 30 00" src="https://user-images.githubusercontent.com/1902323/144210258-5064a453-93c0-4e92-b4e3-3ed14f7c6cd8.png"  width="640" />

#### Desktop

<img src="https://user-images.githubusercontent.com/1902323/144210189-f5352fd6-3b54-47a4-bf2e-5b0bb1c9682d.png" width="640" />

#### iOS

<img src="https://user-images.githubusercontent.com/1902323/143475115-36c7bb6a-7fa3-47c5-b105-545680b9d71f.png" width="400" />

#### Android

<img src="https://user-images.githubusercontent.com/1902323/143475379-ec1c53a2-ceae-4bfa-92b6-1cce6690aeff.png" width="400" />
